core/src/mindustry/world/blocks/production/GenericSmelter.java

```diff
@@ -30,7 +30,15 @@ public void draw(Tile tile){
 
         GenericCrafterEntity entity = tile.ent();
 
-        //draw glowing center
+        //draw topRegion
+        if(entity.warmup > 0f){
+            Draw.color(1f, 1f, 1f, entity.warmup);
+            Draw.rect(topRegion, tile.drawx(), tile.drawy());
+            Draw.color();
+        };
+        
+        
+        //draw "flame" in center
         if(entity.warmup > 0f && flameColor.a > 0.001f){
```

Comment:
you may enclose this `if` into the upper one and get rid of "entity.warmup > 0f &&" here, plus one less check